src/DonationForms/DonationQuery.php

```diff
@@ -83,17 +83,71 @@ public function between($startDate, $endDate)
         return $this;
     }
 
+    /**
+     * @unreleased
+     */
+    public function includeOnlyValidStatuses(): DonationQuery
+    {
+        $this->whereIn('donation.post_status', ['publish', 'give_subscription']);
+
+        return $this;
+    }
+
+    /**
+     * @unreleased
+     */
+    public function includeOnlyCurrentMode(): DonationQuery
+    {
+        $this->joinMeta('_give_payment_mode', 'paymentMode');
+        $this->where('paymentMode.meta_value', give_is_test_mode() ? 'test' : 'live');
+
+        return $this;
+    }
+
     /**
      * Returns a calculated sum of the intended amounts (without recovered fees) for the donations.
+     *
+     * @unreleased Use the NULLIF function to prevent zero values that can generate a wrong final result and use $this->includeOnlyValidStatuses() and $this->includeOnlyCurrentMode()
      * @since 3.12.0
      * @return int|float
      */
-    public function sumIntendedAmount()
+    public function sumIntendedAmount($includeOnlyValidStatuses = true, $includeOnlyCurrentMode = true)
     {
+        if ($includeOnlyValidStatuses) {
+            $this->includeOnlyValidStatuses();
+        }
+
+        if ($includeOnlyCurrentMode) {
+            $this->includeOnlyCurrentMode();
+        }
+
         $this->joinMeta('_give_payment_total', 'amount');
         $this->joinMeta('_give_fee_donation_amount', 'intendedAmount');
         return $this->sum(
-            'COALESCE(intendedAmount.meta_value, amount.meta_value)'
+            'COALESCE(NULLIF(intendedAmount.meta_value,0), NULLIF(amount.meta_value,0))'
+        );
```

Comment:
I think we'll want to add a final fallback in case both values end up being null.

```suggestion
            'COALESCE(NULLIF(intendedAmount.meta_value,0), NULLIF(amount.meta_value,0), 0)'
        );
```

Reply:
Resolved in bb80e91814bd16348ea49368247aacf1aaa2abf4